### PR TITLE
dev: improving the handling of arguments within the script

### PR DESCRIPTION
**Pull request by rkachach**
So far all the arguments were being handled as separate getopts flags introducing a lot of boilerplate code for every new flag. The new approach improves the arguments handling by using env variables.

This is a followup PR to address the improvements proposed on https://github.com/rook/rook/pull/13298#pullrequestreview-1768437705

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: tests/scripts/create-dev-cluster.sh [hunk 1]
```diff
@@ -4,35 +4,39 @@ DEFAULT_NS="rook-ceph"
 CLUSTER_FILES="common.yaml operator.yaml cluster-test.yaml cluster-on-pvc-minikube.yaml dashboard-external-http.yaml toolbox.yaml"
 MONITORING_FILES="monitoring/prometheus.yaml monitoring/service-monitor.yaml monitoring/exporter-service-monitor.yaml monitoring/prometheus-service.yaml monitoring/rbac.yaml"
 SCRIPT_ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd -P)
-ROOK_EXAMPLES_DIR="${SCRIPT_ROOT}/../../deploy/examples/"
+
+# Script arguments: new arguments must be added here (following the same format)
+export ROOK_PROFILE_NAME="${ROOK_PROFILE_NAME:=rook}" ## Specify the minikube profile name
+export ROOK_CLUSTER_NS="${ROOK_CLUSTER_NS:=$DEFAULT_NS}" ## CephCluster namespace
+export ROOK_OPERATOR_NS="${ROOK_OPERATOR_NS:=$DEFAULT_NS}" ## Rook operator namespace (if different from CephCluster namespace)
+export ROOK_EXAMPLES_DIR="${EXAMPLES_DIR:=deploy/examples}" ## Path to Rook examples directory (i.e github.com/rook/rook/deploy/examples)
```
Comment (sp98) [inline tests/scripts/create-dev-cluster.sh:12]: just for consistency..

```suggestion
export ROOK_EXAMPLES_DIR="${ROOK_EXAMPLES_DIR:=deploy/examples}" ## Path to Rook examples directory (i.e github.com/rook/rook/deploy/examples)
```

**Review by sp98 — CHANGES_REQUESTED** [review]
nit

File: tests/scripts/create-dev-cluster.sh (hunk 1 above)
Comment (rkachach) [reply tests/scripts/create-dev-cluster.sh]: fixed

**Review by BlaineEXE — APPROVED** [review]

**Review by sp98 — APPROVED** [review]

File: tests/scripts/create-dev-cluster.sh [hunk 2]
```diff
@@ -208,21 +213,6 @@ while getopts ":hrmfd:p:i:c:o:" opt; do
...
-	    minikube_profile_name="$OPTARG"
-	    ;;
-	i)
-	    cluster_spec_file="$OPTARG"
-	    ;;
-	c)
-	    rook_cluster_ns="$OPTARG"
-	    ;;
-	o)
-	    rook_operator_ns="$OPTARG"
```
Comment (parth-gr) [inline tests/scripts/create-dev-cluster.sh:224]: we cant customise namesapce now?

**Pull request merged**